### DISCUSSION TITLE
[Support]: Error trying to update Frigate on Debian 12 with compose

### DISCUSSION THREAD

**Discussion by Freezerz** [blakeblackshear/frigate#10942] [General Support] [answered]
### Describe the problem you are having

When I run the following command I get the below error message.
`$ sudo docker compose pull`

> WARN[0000] /home/compose_frigate/docker-compose.yml: `version` is obsolete 
> [+] Pulling 7/8
>  ⠋ frigate 7 layers [⣿⣿⣿⣿⣿⣿⣿]      0B/0B      Pulling                                                                     1001.1s 
>    ✔ 5d0aeceef7ee Pull complete                                                                                            140.1s 
>    ✔ cceb31c3be64 Pull complete                                                                                            985.1s 
>    ✔ 3cecf6de71f0 Download complete                                                                                        800.9s 
>    ✔ 4c1dd90319df Download complete                                                                                        262.7s 
>    ✔ 66ca6c6d6804 Download complete                                                                                        263.4s 
>    ✔ 63ccd149174d Download complete                                                                                        263.7s 
>    ✔ d7349a098d45 Download complete                                                                                        273.7s 
> failed to register layer: write /usr/local/lib/python3.9/dist-packages/openvino/libs/libopenvino_intel_cpu_plugin.so: no space left on device

Edit: I'm sorry I forgot to mention that it's not a space issue:

`$ df -h /usr/local/lib/`
> Filesystem                  Type  Size  Used Avail Use% Mounted on
> /dev/mapper/skynet--vg-root ext4   23G  5,3G   17G  25% /

The OS is Debian GNU/Linux 12 (bookworm) and the python version is 3.11.

What am I doing wrong? I have previously had success updating Frigate with the exact same command.

Do I have to somehow downgrade python to 3.9 in order to update Frigate or is there another way?

### Version

0.13.1-34fb1c2

### Frigate config file

```yaml
N/A
```

### Relevant log output

```shell
N/A
```

### FFprobe output from your camera

```shell
N/A
```

### Operating system

Debian

### Install method

Docker Compose

### Coral version

USB

### Network connection

Wired

### Camera make and model

N/A

**Comment by NickM-27** [comment]
> failed to register layer: write /usr/local/lib/python3.9/dist-packages/openvino/libs/libopenvino_intel_cpu_plugin.so: no space left on device

you are out of storage space

**Reply by Freezerz** [reply]
I'm sorry I forgot to mention that it's not a space issue (17G is available):

`$ df -h /usr/local/lib/`
> Filesystem                  Type  Size  Used Avail Use% Mounted on
> /dev/mapper/skynet--vg-root ext4   23G  5,3G   17G  25% /

**Reply by NickM-27** [reply]
docker is clear that it is a space issue, maybe you have configured docker to only be allowed to use a certain % of total system storage (or maybe it has a default config that way)

In any case, this is a docker issue and nothing to do with frigate

**Reply by NickM-27** [reply]
you can try running docker image prune to remove dangling images

**Reply by Freezerz** [reply, marked as answer]
Thank you for pointing me in the right direction. I had the impression it was an issue with trying to use python3.9 when only python3.11 was installed. But in fact it was a storage issue.

My /var partition ran out of space during the update. I don't know if a temporary symlink is created from /usr/local/lib/python3.9 (/ partition) to somewhere inside /var (/var partition) or if the error message is just a bit confusing.

I freed up space on /var using apt-get autoclean ; apt-get clean and then the update was successful.
